collector/src/main.rs

```diff
@@ -745,137 +686,196 @@ fn main() {
     }
 }
 
-fn main_result() -> anyhow::Result<i32> {
-    env_logger::init();
+#[derive(Debug, clap::Parser)]
+#[clap(about, version, author)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Commands,
+}
 
-    let matches = clap_app!(rustc_perf_collector =>
-        (version: "0.1")
-        (author: "The Rust Compiler Team")
-        (about: "Collects Rust performance data")
+#[derive(Debug, clap::Args)]
+struct RustcOption {
+    /// The path to the local rustc to benchmark
+    // Not a `PathBuf` because it can be a file path *or* a `+`-prefixed
+    // toolchain name, and `PathBuf` doesn't work well for the latter.
+    rustc: String,
+}
 
-         // For each subcommand we list the mandatory arguments in the required
-         // order, followed by the options in alphabetical order.
+#[derive(Debug, clap::Args)]
+struct IdOption {
+    /// Identifier to associate benchmark results with
+    id: String,
+}
 
-        (@subcommand bench_local =>
-            (about: "Benchmarks a local rustc")
+#[derive(Debug, clap::Args)]
+struct LocalOptions {
+    /// Measure the build profiles in this comma-separated list
+    // This must be normalized via `ProfilerKind::expand_all()` before use.
+    #[clap(
+        long = "builds",
+        arg_enum,
+        multiple_values = true,
+        use_delimiter = true,
+        require_delimiter = true,
+        // Don't run rustdoc by default
+        default_value = "Check,Debug,Opt",
```

Comment:
Have you tested this?  I would have assumed you'd need to do `default_values = ["Check", "Debug", "Opt"]` (ie the default is populated after doing the splitting)

Reply:
I definitely tested it :)  I assumed `default_value` requires a value that would be valid command line syntax. E.g. I can do `--builds Check,Debug,Opt` or `--builds=Check,Debug,Opt`.

Reply:
Surprising, we special cased this in clap.
```rust
        fn process_default_vals(arg: &Arg<'_>, default_vals: &[&OsStr]) -> Vec<OsString> {
            ...
                    for val in val.split(delim) {
                        vals.push(val.to_os_str().into_owned());
                    }
```